api/hypershift/v1beta1/hostedcluster_types.go

```diff
@@ -1694,19 +1694,73 @@ type AWSServiceEndpoint struct {
 }
 
 type AzurePlatformSpec struct {
+	// Credentials is the object containing the Azure credentials needed when creating and managing Azure cloud
+	// infrastructure resources
+	//
+	// +kubebuilder:validation:Required
 	Credentials corev1.LocalObjectReference `json:"credentials"`
-	// The cloud environment identifier, valid values could be found here: https://github.com/Azure/go-autorest/blob/4c0e21ca2bbb3251fe7853e6f9df6397f53dd419/autorest/azure/environments.go#L33
-	// +kubebuilder:validation:Enum=AzurePublicCloud;AzureUSGovernmentCloud;AzureChinaCloud;AzureGermanCloud
+
+	// Cloud is the cloud environment identifier, valid values could be found here: https://github.com/Azure/go-autorest/blob/4c0e21ca2bbb3251fe7853e6f9df6397f53dd419/autorest/azure/environments.go#L33
+	//
+	// +kubebuilder:validation:Enum=AzurePublicCloud;AzureUSGovernmentCloud;AzureChinaCloud;AzureGermanCloud;AzureStackCloud
 	// +kubebuilder:default="AzurePublicCloud"
-	Cloud             string `json:"cloud,omitempty"`
-	Location          string `json:"location"`
+	Cloud string `json:"cloud,omitempty"`
+
+	// Location is Azure region in which the Azure cloud infrastructure resources will be created
+	//
+	// +kubebuilder:validation:Required
+	// +kubebuilder:validation:XValidation:rule="self == oldSelf",message="Location is immutable"
+	// +immutable
+	Location string `json:"location"`
+
+	// ResourceGroupName is the name of the resource group where any cloud infrastructure resources
+	// created by the Hosted Cluster are to be placed
+	//
+	// +kubebuilder:default:=default
+	// +kubebuilder:validation:Required
+	// +kubebuilder:validation:XValidation:rule="self == oldSelf",message="ResourceGroupName is immutable"
+	// +immutable
 	ResourceGroupName string `json:"resourceGroup"`
-	VnetName          string `json:"vnetName"`
-	VnetID            string `json:"vnetID"`
-	SubnetName        string `json:"subnetName"`
-	SubscriptionID    string `json:"subscriptionID"`
-	MachineIdentityID string `json:"machineIdentityID"`
-	SecurityGroupName string `json:"securityGroupName"`
+
+	// VnetName is the name of the VNET to place created subnets and VMs into
+	//
+	// +kubebuilder:validation:Required
+	// +kubebuilder:validation:XValidation:rule="self == oldSelf",message="VnetName is immutable"
+	// +immutable
+	VnetName string `json:"vnetName"`
+
+	// VnetID is the ID of the VNET to place created subnets and VMs into
+	//
+	// +kubebuilder:validation:XValidation:rule="self == oldSelf",message="VnetID is immutable"
+	// +optional
```

Comment:
> If no value is passed, nothing should happen. This value is passed to CAPZ to consume when creating the VMs

I am not sure I understand this. What is the effect of "nothing should happen" on the deployed cluster? the VMs have to be placed in some Azure Virtual Networks. What are VMs in this context? does it impact NodePool VMs?

What is its difference with `VnetName`? We need to understand this because a Vnet will possibly be provided by the end-user, which will belong to a different resource group name that other resources that will be created by us. Depending on the restrictions with the attributes in here this might have an impact on the solution.

This is also related to the question in https://github.com/openshift/hypershift/pull/3879/files#r1577438805.